Client/Frontend/Browser/QRCodeViewController.swift

```diff
@@ -0,0 +1,265 @@
+    func goBack() {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func openLight() {
+        if isLightOn {
+            do {
+                try captureDevice.lockForConfiguration()
+                captureDevice.torchMode = AVCaptureTorchMode.off
+                captureDevice.unlockForConfiguration()
+                navigationItem.rightBarButtonItem?.image = UIImage(named: "qrcode-light")
+                navigationItem.rightBarButtonItem!.tintColor = UIColor.white
+            } catch {
+                print(error)
+            }
+        } else {
+            do {
+                try captureDevice.lockForConfiguration()
+                captureDevice.torchMode = AVCaptureTorchMode.on
+                captureDevice.unlockForConfiguration()
+                navigationItem.rightBarButtonItem?.image = UIImage(named: "qrcode-isLighting")
+                navigationItem.rightBarButtonItem?.tintColor = QRCodeViewControllerUX.isLightingNavigationItemColor
+            } catch {
+                print(error)
+            }
+        }
+        isLightOn = !isLightOn
+    }
+    
+    func setupCamera() {
+        do {
+            let input = try AVCaptureDeviceInput(device: captureDevice)
+            captureSession.addInput(input)
+        } catch {
+            print(error)
+        }
+        
+        let output = AVCaptureMetadataOutput()
+        if captureSession.canAddOutput(output) {
+            captureSession.addOutput(output)
+            output.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            output.metadataObjectTypes = [AVMetadataObjectTypeQRCode]
+        }
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer?.videoGravity = AVLayerVideoGravityResizeAspectFill
+        previewLayer?.frame = UIScreen.main.bounds
+        view.layer.addSublayer(previewLayer!)
+        
+        captureSession.startRunning()
+    }
+    
+    // MARK: - AVCaptureMetadataOutputObjectsDelegate
+    func captureOutput(_ captureOutput: AVCaptureOutput!, didOutputMetadataObjects metadataObjects: [Any]!, from connection: AVCaptureConnection!) {
+        if metadataObjects == nil || metadataObjects.count == 0 {
+            self.captureSession.stopRunning()
+            let alert = UIAlertController(title: "", message: NSLocalizedString("The data is invalid", comment: "Text of the prompt user the data is invalid"), preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: NSLocalizedString("OK", comment: "OK button"), style: .default, handler: { (UIAlertAction) in
+                self.captureSession.startRunning()
+            }))
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            self.captureSession.stopRunning()
+            stopScanLineAnimation()
+            let metaData = metadataObjects.first as! AVMetadataMachineReadableCodeObject
+            let url = NSURL(string: metaData.stringValue)
```

Comment:
I want to test this case, and this case can only build on iPhone or iPad, but I found that I was deleted form the apple developer group.
So could you help to add lyang@mozilla.com to the apple developer group and test group?